### PR TITLE
Infinispan needs

### PR DESCRIPTION
**Pull request by Sanne**
A couple of improvements noticed while synching up Infinispan to our latest version.

 - https://hibernate.atlassian.net/browse/HSEARCH-2872
 - https://hibernate.atlassian.net/browse/HSEARCH-2873

### CODE REVIEW

File: engine/src/main/java/org/hibernate/search/spi/impl/DelegatingIndexedTypeMap.java [hunk 1]
```diff
@@ -29,6 +29,7 @@
 
 	private final Map<IndexedTypeIdentifier,V> map;
 	private final Map<String,IndexedTypeIdentifier> name2keyMapping;
+	private volatile IndexedTypeSet keyset;
```
Comment (yrodiere) [inline engine/src/main/java/org/hibernate/search/spi/impl/DelegatingIndexedTypeMap.java]: I don't understand, how is this related to removing the "put" method? Seems it isn't used, either.

Comment (Sanne) [reply engine/src/main/java/org/hibernate/search/spi/impl/DelegatingIndexedTypeMap.java]: Indeed, it's part of a WIP I separated. Forgot to remove this piece.

FYI the (unrelated) work is that we use `keySet` quite regularly and it's not efficient to rebuild the returned `IndexedTypeSet` each time. I don't like the volatile path though, so more changes will be needed to fix that.

thanks for catching that!

**Comment by Sanne** [comment]
merged
